lib/topics_bulk_action.rb

```diff
@@ -271,7 +271,7 @@ def append_tags
   def remove_tags
     topics.each do |t|
       if guardian.can_edit?(t)
-        TopicTag.where(topic_id: t.id).delete_all
+        TopicTag.where(topic_id: t.id).in_batches.destroy_all
```

Comment:
Since we're now loading records into memory, slap an `#in_batches` on there for good measure.